### ISSUE TITLE
Rails 6 support

### ISSUE THREAD

**Issue by jrochkind** [jrochkind/bento_search#39]
We could probably get tests passing on Rails 6 with very few changes, but tests passing can hide some actual difficulties. 

Rails 6 by default does not use sprockets asset pipeline for Javascript (it does use it for CSS), but we provide JS in such a way for asset pipeline.  The JS for ajax loading of search results is a pretty key part of this gem. 

We can get tests passing anyway by modifying our in-repo dummy app to have sprockets compile JS, and have the layout use a sprockets-compiled application.js.  But what do we expect users to do? They're going to be lost. 

We could provide instructions for how to get JS asset pipepline working in Rails 6, and say, sorry, that's the way this gem supports using it's JS, it's all we support at present. 

That probably does make sense as at least an initial step, that could be released without even a major version bump.  I don't think we should change gemspec to allow Rails 6 without those instructions in README. 

Ultimately we "should" provide our JS as npm package instead for use with webpacker... but I am still not really happy with the developer ergonomics of this for a Rails engine, at least not with what i've figured out so far. (One idea is using yarn's ability to point to an npm package from a git repo to point it directly to bentosearch repo with a version tag?)  Probably for the future.  

Wait alternately though -- our JS is actually only 20 lines in one file! It's not necessarily crazy to just copy it into the local app... our JS also depends on JQuery, would be nice to have a pure javascript version.

**Comment by dkinzer** [comment]
Just an FYI, we're giving Rails 6 a try now that BL is officially supporting it and we forked this repo to allow Rails 6.  We ran it successfully locally.

**Comment by jrochkind** [comment]
Thanks @dkinzer! Are you still including the JS with sprockets? Want to share your local fork here? Do you have an idea of what it would take to get Rails 6 support official here?

**Comment by dkinzer** [comment]
This is dumb, but I didn't even know bento_search had AJAX capabilities.   We don't use them (in fact we've sort of reinvented the wheel here for a new feature we're developing).

Currently we are using both sprockets and webpacker for JS stuff.  We'll eventually move all our JS to webpacker.  But, for now we have an unsavory mix of the two.

> Want to share your local fork here? Do you have an idea of what it would take to get Rails 6 support official here?
Assuming the only issue is that rails 6 uses webpacker by default, I don't know.  But,  I like BL's approach of just giving the user the option of using either or, and providing instructions for both:
https://github.com/projectblacklight/blacklight#using-the-javascript

**Comment by jrochkind** [comment]
>  But, I like BL's approach of just giving the user the option of using either or, and providing instructions for both:

Absolutely, but we're not building/releasing an npm package right now for use with webpacker. Need to figure out how to do that in a low-burden maintainable way for the long term. I'll get around to it eventually.   Also think I have an idea involving yarn pointing directly at a git repo for an npm package that might be even lower-maintenance. 

In the meantime... I suppose we could go ahead and make the bento_search `gemspec` *allow* Rails 6, with big warnings about you are on your own figuring out how to get the JS to work... except, doh, I just remembered, the problem is that our CI won't pass (at least the parts of the CI that use JS!) until we solve it somehow! 

Thus stuck at this impasse for now. I assume you aren't running CI on your fork? 

I'll again ask if you're willing to share the location of your local fork here, so we can see what you've changed (anything except the gemspec), so it's available for anyone else that comes along and is curious? If not, up to you! 

And nothing wrong with re-inventing the wheel for the JS, it's meant to support you doing that if you find it more convenient than the built-in JS. The built-in JS is actually using techniques pretty similar to all this new-fangled "stimulus" stuff Rails is adding -- but not actually doing it with stimulus!

**Comment by jrochkind** [comment]
And I just noticed CI isn't running at all right now! #41 

I am not currently using bento_search myself, and my support time is limited (although I still will try to find time to keep the lights on, for sentimental reasons!). 

But if you or anyone else were looking for something to do to help with supporting... getting CI running on github actions doing the same thing it used to do on travis would be one thing.

**Comment by dkinzer** [comment]
> Absolutely, but we're not building/releasing an npm package right now for use with webpacker. Need to figure out how to do that in a low-burden maintainable way for the long term.

Hmm, looks like `yarn` (v2) may allow you to add js packages from github repository subfolders. 

Ref: https://github.com/yarnpkg/yarn/issues/4725#issuecomment-675649873
